src/plugincollection.js

```diff
@@ -208,6 +208,21 @@ export default class PluginCollection {
 	}
 
 	/**
+	 * Destroys all loaded plugins.
+	 *
+	 * @returns {Promise}
+	 */
+	destroy() {
+		const promises = [];
+
+		for ( const [ , pluginInstance ] of this ) {
```

Comment:
We cannot use `Array.from( this._plugins.values() )` because we will call `plugin.destroy()` twice for plugins that have a defined name.